sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/CodeGenerator.scala

```diff
@@ -1004,7 +1016,8 @@ object CodeGenerator extends Logging {
    * weak keys/values and thus does not respond to memory pressure.
    */
   private val cache = CacheBuilder.newBuilder()
-    .maximumSize(100)
+    .maximumWeight(10 * 1024 * 1024)
```

Comment:
Hi, @viirya . If this is configured as 10MB, we can have 160 cached items with each 64KB code.
It looks like this patch will make Spark consume more memory than the existing one.